cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -1238,6 +1239,25 @@ if(imt AND NOT builtin_tbb)
       set(builtin_tbb ON CACHE BOOL "Enabled because imt is enabled, but TBB was not found" FORCE)
     endif()
   endif()
+
+  # Check that the found TBB does not use captured exceptions.
+  if(TBB_FOUND)
+    set(CMAKE_REQUIRED_INCLUDES "${TBB_INCLUDE_DIRS}")
+    check_cxx_source_compiles("
+#include <tbb/tbb_config.h>
```

Comment:
oneTBB 2021.2 does not have this header.

(found out in https://github.com/cms-sw/cmsdist/pull/6936)

Reply:
Thanks for testing and reporting. Indeed, oneTBB doesn't have the header anymore nor the "captured exceptions" thing, so https://github.com/root-project/root/pull/8239 skips the test under those conditions.

@oshadura our `FindTBB.cmake` cannot detect versions of newer TBBs because, guess what, they don't have the header `tbb/tbb_stddef.h` either. I was first checking `TBB_VERSION`, but that's empty in such case. Probably somebody with proper CMake skills should take a look and make the detection work properly...

Reply:
@hahnjo thanks a lot for fix! could you please open a GH issue/improvement to revisit `FindTBB.cmake`?